consensus/src/pipeline/header_processor/post_pow_validation.rs

```diff
@@ -69,7 +70,21 @@ impl HeaderProcessor {
         ctx: &mut HeaderProcessingContext,
         header: &Header,
     ) -> BlockProcessResult<()> {
-        // TODO: Implement this
+        let expected_block_parents = self.parents_manager.calc_block_parents(ctx.pruning_point, header.direct_parents());
+        if !expected_block_parents.iter().enumerate().all(|(block_level, expected_level_parents)| {
+            let header_level_parents = &header.parents_by_level[block_level];
```

Comment:
You must first verify that `block_level <  &header.parents_by_level.len()`. Perhaps extract the equality test to a method and compare number of levels first